### ISSUE TITLE
Installation to Libvirt fails due to no space left in /run

### ISSUE THREAD

**Issue by noseka1** [openshift/okd#649]
**Describe the bug**

During the cluster bootstrap, the command `release-image-download.sh` running on the bootstrap node fails due to no space left on device. The tmpfs mounted on `/run` is full:

```
$ df -kh /run
Filesystem      Size  Used Avail Use% Mounted on
tmpfs           786M  786M     0 100% /run
```

For some reason, the size of `/run` volume is only 786M. That is not enough for storing `/run/mco-machine-os-content/os-content-232915737` image which has around 1.1 GiB.

Older OKD versions don't have this problem. They mount 2.0 GiB volume on `/run`.

```
$ df -kh /run
Filesystem      Size  Used Avail Use% Mounted on
tmpfs           2.0G  948K  2.0G   1% /run
```

**Version**

Versions affected by this issue:
4.7.0-0.okd-2021-05-22-050008
4.7.0-0.okd-2021-03-28-152009
4.7.0-0.okd-2021-03-21-094146

Working versions (not affected by this issue):
4.7.0-0.okd-2021-03-07-090821
4.7.0-0.okd-2021-02-25-144700

**How reproducible**

Use CodeReady Containers https://github.com/code-ready/snc to deploy a cluster.

**Log bundle**

Related logs obtained using `journalctl -b -f -u release-image.service -u bootkube.service`:

```
...

May 14 23:57:10 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:10.077014    1845 run.go:18] Running: nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215
May 14 23:57:10 crc-hphs7-bootstrap release-image-download.sh[1845]: ionice: failed to execute oc: No such file or directory
May 14 23:57:10 crc-hphs7-bootstrap release-image-download.sh[1845]: W0514 23:57:10.080897    1845 run.go:44] nice failed: running nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35f
bbc9215 failed: ionice: failed to execute oc: No such file or directory
May 14 23:57:10 crc-hphs7-bootstrap release-image-download.sh[1845]: : exit status 127; retrying...
May 14 23:57:15 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:15.081266    1845 run.go:18] Running: nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215
May 14 23:57:15 crc-hphs7-bootstrap release-image-download.sh[1845]: ionice: failed to execute oc: No such file or directory
May 14 23:57:15 crc-hphs7-bootstrap release-image-download.sh[1845]: W0514 23:57:15.091356    1845 run.go:44] nice failed: running nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35f
bbc9215 failed: ionice: failed to execute oc: No such file or directory
May 14 23:57:15 crc-hphs7-bootstrap release-image-download.sh[1845]: : exit status 127; retrying...
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:25.091785    1845 run.go:18] Running: nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: ionice: failed to execute oc: No such file or directory
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: W0514 23:57:25.097418    1845 run.go:44] nice failed: running nice -- ionice -c 3 oc image extract --path /:/run/mco-machine-os-content/os-content-805812742 quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35f
bbc9215 failed: ionice: failed to execute oc: No such file or directory
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: : exit status 127; retrying...
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:25.097452    1845 update.go:371] Falling back to using podman cp to fetch OS image content
May 14 23:57:25 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:25.097480    1845 run.go:18] Running: nice -- ionice -c 3 podman pull -q quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215
May 14 23:57:26 crc-hphs7-bootstrap podman[1854]: 2021-05-14 23:57:26.915382754 +0000 UTC m=+1.798203633 image pull
May 14 23:57:26 crc-hphs7-bootstrap release-image-download.sh[1845]: cb79cfe99ac5709fdb0ea42fc6fb95a558b0f7969603fafd9a1a35ec7c9bca08
May 14 23:57:26 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:26.922930    1845 rpm-ostree.go:261] Running captured: podman create --net=none --annotation=org.openshift.machineconfigoperator.pivot=true --name ostree-container-pivot-383585c8-8705-41ee-845c-0e4a3d0848b1 quay.io/openshift/okd-content@sh
a256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215
May 14 23:57:27 crc-hphs7-bootstrap podman[1883]: 2021-05-14 23:57:27.018023819 +0000 UTC m=+0.082075206 container create 2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7 (image=quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215, name=ostree-container-p
ivot-383585c8-8705-41ee-845c-0e4a3d0848b1, architecture=x86_64, vcs-ref=b632cfe7be5dc185b8259395daacc82922178fb8, io.openshift.build.commit.id=b632cfe7be5dc185b8259395daacc82922178fb8, io.openshift.build.name=, com.redhat.license_terms=https://www.redhat.com/agreements, com.redhat.component=openshift-enterprise-base-co
ntainer, io.openshift.expose-services=, summary=Provides the latest release of Red Hat Universal Base Image 8., io.openshift.build.source-location=https://github.com/openshift/okd-machine-os, vcs-url=https://github.com/openshift/okd-machine-os, io.openshift.build.commit.ref=release-4.7, description=The Universal Base I
mage is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly., mai
ntainer=Red Hat, Inc., com.redhat.build-host=cpt-1004.osbs.prod.upshift.rdu2.redhat.com, io.openshift.build.commit.message=, vcs-type=git, release=202101090039.11723, io.buildah.version=1.16.4, distribution-scope=public, io.k8s.description=This is the base image from which all OpenShift images inherit., io.openshift.bu
ild.commit.author=, io.openshift.build.commit.date=, io.openshift.build.source-context-dir=, build-date=2021-01-09T00:40:49.580557, vendor=Red Hat, Inc., io.openshift.build.namespace=, io.openshift.release.operator=true, name=openshift/ose-base, io.k8s.display-name=OpenShift Base, io.openshift.tags=base rhel8, version=
v4.0, url=https://access.redhat.com/containers/#/registry.access.redhat.com/openshift/ose-base/images/v4.0-202101090039.11723)
May 14 23:57:27 crc-hphs7-bootstrap release-image-download.sh[1845]: I0514 23:57:27.023830    1845 run.go:18] Running: nice -- ionice -c 3 podman cp 2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7:/ /run/mco-machine-os-content/os-content-805812742
May 14 23:57:27 crc-hphs7-bootstrap podman[1914]: 2021-05-14 23:57:27.102788538 +0000 UTC m=+0.064707250 container mount 2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7 (image=quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215, name=ostree-container-pi
vot-383585c8-8705-41ee-845c-0e4a3d0848b1, maintainer=Red Hat, Inc., architecture=x86_64, distribution-scope=public, vcs-ref=b632cfe7be5dc185b8259395daacc82922178fb8, io.openshift.expose-services=, url=https://access.redhat.com/containers/#/registry.access.redhat.com/openshift/ose-base/images/v4.0-202101090039.11723, io
.openshift.build.source-context-dir=, io.k8s.description=This is the base image from which all OpenShift images inherit., io.openshift.release.operator=true, io.openshift.build.commit.author=, com.redhat.license_terms=https://www.redhat.com/agreements, summary=Provides the latest release of Red Hat Universal Base Image
 8., io.openshift.build.source-location=https://github.com/openshift/okd-machine-os, io.openshift.build.name=, io.openshift.tags=base rhel8, version=v4.0, name=openshift/ose-base, vcs-type=git, io.openshift.build.namespace=, io.openshift.build.commit.message=, vendor=Red Hat, Inc., io.openshift.build.commit.date=, vcs-
url=https://github.com/openshift/okd-machine-os, build-date=2021-01-09T00:40:49.580557, io.openshift.build.commit.id=b632cfe7be5dc185b8259395daacc82922178fb8, release=202101090039.11723, com.redhat.component=openshift-enterprise-base-container, com.redhat.build-host=cpt-1004.osbs.prod.upshift.rdu2.redhat.com, io.opensh
ift.build.commit.ref=release-4.7, io.buildah.version=1.16.4, io.k8s.display-name=OpenShift Base, description=The Universal Base Image is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports
 Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.)
May 14 23:57:33 crc-hphs7-bootstrap podman[1914]: 2021-05-14 23:57:33.462166326 +0000 UTC m=+6.424085055 container unmount 2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7 (image=quay.io/openshift/okd-content@sha256:1fc150fd7b47122e3ff344839cd9b2f1085024beb279f6c5eb5c3c35fbbc9215, name=ostree-container-
pivot-383585c8-8705-41ee-845c-0e4a3d0848b1, vcs-ref=b632cfe7be5dc185b8259395daacc82922178fb8, distribution-scope=public, build-date=2021-01-09T00:40:49.580557, io.openshift.build.source-context-dir=, release=202101090039.11723, vendor=Red Hat, Inc., io.k8s.display-name=OpenShift Base, io.k8s.description=This is the bas
e image from which all OpenShift images inherit., io.openshift.build.commit.date=, io.openshift.release.operator=true, io.buildah.version=1.16.4, url=https://access.redhat.com/containers/#/registry.access.redhat.com/openshift/ose-base/images/v4.0-202101090039.11723, name=openshift/ose-base, com.redhat.license_terms=htt
ps://www.redhat.com/agreements, vcs-url=https://github.com/openshift/okd-machine-os, io.openshift.build.commit.author=, io.openshift.build.source-location=https://github.com/openshift/okd-machine-os, io.openshift.tags=base rhel8, io.openshift.build.commit.message=, summary=Provides the latest release of Red Hat Univers
al Base Image 8., com.redhat.build-host=cpt-1004.osbs.prod.upshift.rdu2.redhat.com, architecture=x86_64, maintainer=Red Hat, Inc., io.openshift.build.name=, io.openshift.build.namespace=, io.openshift.expose-services=, com.redhat.component=openshift-enterprise-base-container, description=The Universal Base Image is des
igned and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly., io.openshift.b
uild.commit.id=b632cfe7be5dc185b8259395daacc82922178fb8, version=v4.0, vcs-type=git, io.openshift.build.commit.ref=release-4.7)
May 14 23:57:33 crc-hphs7-bootstrap release-image-download.sh[1845]: time="2021-05-14T23:57:33Z" level=error msg="unable to umount container '2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7': \"error unmounting container 2f4b265b94e0681657b6888afb662b292c23c2f485c3cead0a8aeb32d20bf4b7 root filesystem:
write /var/run/containers/storage/overlay-layers/.tmp-mountpoints.json052892132: no space left on device\""
May 14 23:57:33 crc-hphs7-bootstrap release-image-download.sh[1845]: Error: error copying "/var/lib/containers/storage/overlay/d7e6049e5639fe58ae646ce1a51886b1f94ff53d4a6140f14b3f46d9cb3f5c54/merged" to "/run/mco-machine-os-content/os-content-805812742": Error processing tar file(exit status 1): time="2021-05-14T23:57:
27Z" level=warning msg="ignored xattrs in archive: underlying filesystem doesn't support them" errors="[lsetxattr /extensions/okd/NetworkManager-ovs-1.26.6-1.fc33.x86_64.rpm: operation not supported]"

...
```

**Comment by noseka1** [comment]
Found a simple workaround. When see the "no space left on device" log messages, execute this on the bootstrap node:

```
$ rm -rf mco-machine-os-content
$ ln -s /tmp /run/mco-machine-os-content
```
The image will be saved in /tmp where there is enough space (2 GiB).

**Comment by Littlemoon917** [comment]
Same issue for me on 4.7.0-0.okd-2021-04-24-103438.
I solved by execute:
`$ env TF_VAR_libvirt_bootstrap_memory=8192 openshift-install create cluster`
or when see the "no space left on device" log messages, execute this on the bootstrap node:
`sudo mount -o remount,size=50% tmpfs /run`.
It seems the bootstrap node need more memory after some new version,but the default config not modified at the same time.

**Comment by vrutkovs** [comment]
Sure, we can bump default bootstrap memory in OKD installer

**Comment by cgruver** [comment]
This is exactly what I am running into with CRC builds for OKD 4.7.

I just haven't had time to fix it yet.  Thanks for the `TF_VAR_libvirt_bootstrap_memory` tip @Littlemoon917.

**Comment by pacevedom** [comment]
Shall we take openshift/installer#4964 to master to consider this done?

**Comment by vrutkovs** [comment]
Right, this one needs a PR to master (and a bugzilla so that it can be backported)

**Comment by vrutkovs** [comment]
Fixed in 4.9 nightlies, cherry-picked the fix to 4.8/4.7. Thanks!
